Source/SwiftLintFramework/Rules/Idiomatic/KeyPathExpressionAsFunctionRule.swift

```diff
@@ -0,0 +1,100 @@
+import SourceKittenFramework
+
+public struct KeyPathExpressionAsFunctionRule: ASTRule, OptInRule, ConfigurationProviderRule, AutomaticTestableRule {
+    public var configuration = SeverityConfiguration(.warning)
+
+    public init() {}
+
+    public static let description = RuleDescription(
+        identifier: "key_path_expression_as_function",
+        name: "Key Path Expression as Function",
+        description: "Prefer using key paths instead of closures when possible.",
+        kind: .idiomatic,
+        minSwiftVersion: .fiveDotTwo,
+        nonTriggeringExamples: [
+            Example("let emails = users.map(\\.email)"),
```

Comment:
Do you think it would read better if we used raw string literals here?

```suggestion
            Example(#"let emails = users.map(\.email)"#),
```